### PR TITLE
TASK-58023: Add a SecurityCheckAuthentication Plugin

### PR DESCRIPTION
**Pull request by hakermi**
This PR introduces  a new plugin called SecurityCheckAuthenticationPlugin wich will be used in the OrganizationAuthenticater to create some logic checks during the athentication phase, when the authentication fails
and when the athentication suceeds , which will be done by its three genric functions (doCheck, onCheckFail, onCheckSuccess).

### CODE REVIEW

**Review by ahamdi — APPROVED** [review]

File: exo.core.component.organization.api/src/main/java/org/exoplatform/services/organization/auth/SecurityCheckAuthenticationPlugin.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
+package org.exoplatform.services.organization.auth;
+
+import org.exoplatform.container.component.BaseComponentPlugin;
+import org.exoplatform.services.organization.User;
+
+public abstract class SecurityCheckAuthenticationPlugin extends BaseComponentPlugin {
```
Comment (boubaker) [inline exo.core.component.organization.api/src/main/java/org/exoplatform/services/organization/auth/SecurityCheckAuthenticationPlugin.java:6]: Why adding this plugin in `core` project level if it's not used inside, neither referenced? In addition, you miss the Header license in fact.

Comment (hakermi) [reply exo.core.component.organization.api/src/main/java/org/exoplatform/services/organization/auth/SecurityCheckAuthenticationPlugin.java:6]: @boubaker , I followed the same replacement location with the Authenticaterplugin, its' also  not used inside the core module, neither referenced

Comment (boubaker) [reply exo.core.component.organization.api/src/main/java/org/exoplatform/services/organization/auth/SecurityCheckAuthenticationPlugin.java:6]: Not same in fact, it's used here: https://github.com/Meeds-io/core/commit/9953ec25b8427bb02c87d3427b7600640ad989ab#diff-9b007092c59894b68a04cc614b92801bbdb4ab92cdf2dba7ec60b4b133d8d844R76

And especially referenced in Default implementation here : 
https://github.com/Meeds-io/core/commit/9953ec25b8427bb02c87d3427b7600640ad989ab#diff-9b007092c59894b68a04cc614b92801bbdb4ab92cdf2dba7ec60b4b133d8d844R219-R230

Comment (hakermi) [reply exo.core.component.organization.api/src/main/java/org/exoplatform/services/organization/auth/SecurityCheckAuthenticationPlugin.java:6]: header added : https://github.com/Meeds-io/core/commit/ceced312a7048b228e1b86b9d28c3bd33efae30d#diff-174ccd71867b79cfb62756ad4723a3c0d14c4805309f8ae6a625e65e36cdf4bfR1-R16

Ok i see, where do you suggest to add it ?

Comment (boubaker) [reply exo.core.component.organization.api/src/main/java/org/exoplatform/services/organization/auth/SecurityCheckAuthenticationPlugin.java:6]: if it's used in gatein-portal implementation only, it should goes there, else adding it inside an API/Default implementation here & used elsewhere will make it more complicated for maintenance & evolutivity.

(BTW, you have a sonar check fail)

Comment (hakermi) [reply exo.core.component.organization.api/src/main/java/org/exoplatform/services/organization/auth/SecurityCheckAuthenticationPlugin.java:6]: Ok, i will proceed to do this, i will add it in gaten-portal

Comment (hakermi) [reply exo.core.component.organization.api/src/main/java/org/exoplatform/services/organization/auth/SecurityCheckAuthenticationPlugin.java:6]: @boubaker added in this PR: https://github.com/Meeds-io/gatein-portal/pull/379
i will remove this commit from the FB

**Pull request merged**